mlflow/pyfunc/__init__.py

```diff
@@ -2429,12 +2429,16 @@ def predict(model_input: List[str]) -> List[str]:
                 CHAT_MODEL_INPUT_SCHEMA,
                 CHAT_MODEL_OUTPUT_SCHEMA,
             )
-            input_example = CHAT_MODEL_INPUT_EXAMPLE
+            input_example = input_example or CHAT_MODEL_INPUT_EXAMPLE
 
-            # perform output validation and throw if
-            # output is not coercable to ChatResponse
-            messages = [ChatMessage(**m) for m in input_example["messages"]]
-            params = ChatParams(**{k: v for k, v in input_example.items() if k != "messages"})
+            if isinstance(input_example, list) and isinstance(input_example[0], ChatMessage):
+                example = {"messages": [each_message.to_dict() for each_message in input_example]}
+                params = ChatParams(**{k: v for k, v in example.items() if k != "messages"})
```

Comment:
What's the usage here? Seems example only contains "messages" key and params will always be empty

Reply:
Added an test example that users could use and verify that.
```
params_subset = {
        "max_tokens": 100,
    }
    input_example = {
        "messages": [
            {
                "role": "user",
                "content": "What is Retrieval-augmented Generation?",
            }
        ],
        **params_subset,
    }
    ```